### PR TITLE
feat(GitHub Node): Add workflow resource operations

### PR DESCRIPTION
**Pull request by idanfishman**
## Summary

Add the ability to use GitHub workflows as Nodes:
- Disable a workflow
- Dispatch a workflow event
- Enable a workflow
- Get a workflow
- Get the usage of a workflow
- List workflows

Following [GitHub's API](https://docs.github.com/en/rest/actions/workflows?apiVersion=2022-11-28).

It enhances the integration between n8n and GitHub Actions Workflows, enabling greater automation by using GitHub nodes instead of HTTP nodes.

Screenshot 2024-09-09 at 20 03 31

## Review / Merge checklist

- [X] PR title and summary are descriptive. ([conventions](../blob/master/.github/pull_request_title_conventions.md)) <!--
- [X] [Docs updated](https://github.com/n8n-io/n8n-docs) or follow-up ticket created (https://github.com/n8n-io/n8n-docs/pull/2463)
- [ ] Tests included.
- [ ] PR Labeled with `release/backport` (if the PR is an urgent fix that needs to be backported)

### CODE REVIEW

**Comment by Joffcom** [comment]
Hey @idanfishman,

Thanks for the PR, We have created "GHC-221" as the internal reference to get this reviewed.

One of us will be in touch if there are any changes needed, in most cases this is normally within a couple of weeks but it depends on the current workload of the team.

**Comment by Joffcom** [comment]
Hey @idanfishman,

Thanks for the PR, We have created "GHC-221" as the internal reference to get this reviewed.

One of us will be in touch if there are any changes needed, in most cases this is normally within a couple of weeks but it depends on the current workload of the team.

**Comment by Joffcom** [comment]
Hey @idanfishman,

Thanks for the PR, We have created "GHC-221" as the internal reference to get this reviewed.

One of us will be in touch if there are any changes needed, in most cases this is normally within a couple of weeks but it depends on the current workload of the team.

**Comment by idanfishman** [comment]
> Hey @idanfishman,
> 
> Thanks for the PR, We have created "GHC-221" as the internal reference to get this reviewed.
> 
> One of us will be in touch if there are any changes needed, in most cases this is normally within a couple of weeks but it depends on the current workload of the team.

Any progress?

File: packages/nodes-base/nodes/Github/Github.node.ts [hunk 1]
```diff
@@ -15,8 +15,8 @@ import {
 	githubApiRequestAllItems,
 	isBase64,
 } from './GenericFunctions';
-
-import { getRepositories, getUsers } from './SearchFunctions';
+import { getRepositories, getUsers, getWorkflows } from './SearchFunctions';
+import { validateJSON } from '../ClickUp/GenericFunctions';
```
Comment (Joffcom) [inline packages/nodes-base/nodes/Github/Github.node.ts:19]: Is this needed if the same function has been added to the GenericFunctions?

Comment (idanfishman) [reply packages/nodes-base/nodes/Github/Github.node.ts:19]: So basically if it was part of the generic functions i would import it from there

Comment (idanfishman) [reply packages/nodes-base/nodes/Github/Github.node.ts:19]: You're right!

I will change the import path.

**Comment by idanfishman** [comment]
@Joffcom Hey,
I have fixed the import statement, I see that the tests ran successfully can we merge it already?

**Comment by Joffcom** [comment]
Hey @idanfishman,

I plan to do some PR reviews next week to get back on top of them but this will be one of the ones I plan to test and get merged.

**Comment by idanfishman** [comment]
@Joffcom Anything?

**Comment by Joffcom** [comment]
Hey @idanfishman 

It is booked in for this week, sadly with feature changes they get moved in favour of bug fixes.

**Review by Joffcom — APPROVED** [review]
Looks good

**Comment by Joffcom** [comment]
@idanfishman thanks for this, It will be available in the release this week 🙏🏻

**Comment by idanfishman** [comment]
> @idanfishman thanks for this, It will be available in the release this week 🙏🏻

Thanks!

**Comment by janober** [comment]
Got released with [`n8n@1.65.0`](https://github.com/n8n-io/n8n/releases/tag/n8n%401.65.0)

**Pull request merged**